src/Extension/NotifyPaymentStateExtension.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AdeoWeb\SyliusEveryPayPlugin\Extension;
+
+use Payum\Core\Extension\Context;
+use Payum\Core\Extension\ExtensionInterface;
+use Payum\Core\GatewayAwareTrait;
+use Payum\Core\GatewayInterface;
+use Payum\Core\Payum;
+use Payum\Core\Request\GetHttpRequest;
+use Payum\Core\Request\GetStatusInterface;
+use Payum\Core\Request\Notify;
+use Exception;
+
+final class NotifyPaymentStateExtension implements ExtensionInterface
+{
+    use GatewayAwareTrait;
+
+    private const QUERY_PARAM_PAYMENT_REFERENCE = 'payment_reference';
+
+    private bool $isNotify = false;
+
+    public function __construct(
+        private Payum $payum,
+    ) {
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function onPreExecute(Context $context): void
+    {
+
+    }
+
+    public function onExecute(Context $context): void
+    {
+        $request = $context->getRequest();
+
+        if ($this->isNotify) {
+            return;
+        }
+
+        if (!$request instanceof GetStatusInterface || !$this->getPaymentReference($context->getGateway())) {
+            return;
+        }
+
+        $notify = new Notify(null);
+        $context->getGateway()->execute($notify);
+        $this->isNotify = true;
```

Comment:
šitą pagal idėją reikia užsettinti prieš kviečiant execute. nes kiek pamenu tai po notify action'o kažkur patriggerina ir get status action'as, prieš kurį gali vėl šitas extension'as būti iškviestas. o isNotify vis dar bus false, nes vis dar nebus išeita iš 51 eilutės